### PR TITLE
rps-challenge

### PR DESCRIPTION
**Pull request by Christos-Paraskeva**

### CODE REVIEW

File: lib/opponent.rb [hunk 1]
```diff
@@ -0,0 +1,12 @@
+class Opponent
+
+  attr_reader :opponent_move
+
+  def create_opponent
```
Comment (maryrosecook) [inline lib/opponent.rb]: This method appears not to be used.

File: lib/rock_paper_scissors.rb [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+  attr_reader :player, :opponent, :player_choice, :opponent_move
+
+  RULES = {Rock: :Scissors,
+           Paper: :Rock,
+           Scissors: :Paper}
+
+  def initialize(player, opponent=Opponent.new)
+    @player = player
+    @opponent = opponent
+    @opponent_move
```
Comment (maryrosecook) [inline lib/rock_paper_scissors.rb]: Could this line be removed?

File: lib/rock_paper_scissors.rb [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+    @game
+  end
+# opponent makes their move which is then assigned to an instance variable
+# this is then passed to the evaluate_result method which calculates the outcome
+  def outcome
+    @opponent_move = @opponent.move
+    evaluate_result
+  end
+
+  def evaluate_result
```
Comment (maryrosecook) [inline lib/rock_paper_scissors.rb]: Really nice and clear!

File: views/loser.erb [hunk 4]
```diff
@@ -0,0 +1,13 @@
+<div style="width:250px;height:50px;padding:10px;border:10px inset cyan;">
```
Comment (maryrosecook) [inline views/loser.erb]: It feels like the winner, loser and tie templates could be dried up.  Could the code in them be shared? Or could one template be used to show all three game outcomes?

File: README.md [hunk 5]
```diff
@@ -28,65 +13,89 @@ So that I can enjoy myself away from the daily grind
...
+
+If I had more time..
+-------
 
-This repo works with [Coveralls](https://coveralls.io/) to calculate test coverage statistics on each pull request.
+If I had more time, I would have liked to have made the visual interface more fluid and attractive.  While I did make some small changes to it in the end, I feel there wasn't enough time to really make it look and function the way I envisaged given most of my time was spent getting the program to function correctly, which was the priority.  Given I hadn't ever used HTML before, I feel there is a vast wealth of features I could have used to do some really cool things with the visual side of my program.
 
+Similarily, I would have also liked to have had the time to research and learn about implementing more dynamic web pages, which could display visual changes without having to reload the page.  I also would have spent more time refactoring my tests further than I have already done.
```
Comment (Christos-Paraskeva) [inline README.md]: I think, in particular, I could have added a before block before some of the tests that would run through a few of the processes such as the name and weapon selection.  It feels to me there's a way to refactor further continuing on from what the web_helpers file is doing.

File: app.rb [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+
+class Game < Sinatra::Base
+
+  run! if app_file == $0
+
+  get '/' do
+    erb :name
+  end
+# this method calls a class method which create a new instance
+# of the Player class and passes the name as a parameter
```
Comment (Christos-Paraskeva) [inline app.rb]: Noted, regarding the unneccessary comments

**Pull request closed without merging**